lib/cc/engine/analyzers/reporter.rb

```diff
@@ -64,9 +64,17 @@ def mass_threshold
           @mass_threshold ||= language_strategy.mass_threshold
         end
 
+        def base_points
+          @base_points ||= language_strategy.base_points
+        end
+
+        def points_per
+          @points_per ||= language_strategy.points_per
+        end
+
         def new_violation(issue)
           hashes = flay.hashes[issue.structural_hash]
-          Violation.new(language_strategy.base_points, issue, hashes)
+          Violation.new(base_points, points_per, mass_threshold, issue, hashes)
```

Comment:
Since all of these come directly from `language_strategy`, would it be better to just pass the language strategy to `Violation`? The code seems to be saying is that a violation's properties are fairly dependent on the specific language it's from, so that doesn't unreasonable to me.